docs/machine-learning/tutorials/taxi-fare.md

```diff
@@ -39,75 +39,81 @@ This problem is centered around **predicting the fare of a taxi trip in New York
+The **label** is the identifier of the column you want to predict. The identified **features** are used to predict the label.
 
-The **label** is the identifier of the column you are trying to predict. The identified **features** are used to predict the label.
+The provided data set contains the following columns:
 
 * **vendor_id:** The ID of the taxi vendor is a feature.
 * **rate_code:** The rate type of the taxi trip is a feature.
 * **passenger_count:** The number of passengers on the trip is a feature.
-* **trip_time_in_secs:** The amount of time the trip took. You won't know how long the trip takes until after it is completed. You exclude this column from the model.
+* **trip_time_in_secs:** The amount of time the trip took. You want to predict the fare of the trip before the trip is completed. At that moment you don't know how long the trip would take. Thus, the trip time is not a feature and you'll exclude this column from the model.
 * **trip_distance:** The distance of the trip is a feature.
 * **payment_type:** The payment method (cash or credit card) is a feature.
 * **fare_amount:** The total taxi fare paid is the label.
 
-### Create classes and define paths
+## Create data classes
 
-Add the following additional `using` statements to the top of the *Program.cs* file:
+Create classes for the input data and the predictions:
 
-[!code-csharp[AddUsings](../../../samples/machine-learning/tutorials/TaxiFarePrediction/Program.cs#1 "Add necessary usings")]
+1. In **Solution Explorer**, right-click the project, and then select **Add** > **New Item**.
+1. In the **Add New Item** dialog box, select **Class** and change the **Name** field to *TaxiTrip.cs*. Then, select the **Add** button.
+1. Add the following `using` statements to the new file:
+
+   [!code-csharp[AddUsings](../../../samples/machine-learning/tutorials/TaxiFarePrediction/TaxiTrip.cs#1 "Add necessary usings")]
+
+Remove the existing class definition and add the following code, which has two classes `TaxiTrip` and `TaxiTripFarePrediction`, to the *TaxiTrip.cs* file:
+
+[!code-csharp[DefineTaxiTrip](../../../samples/machine-learning/tutorials/TaxiFarePrediction/TaxiTrip.cs#2 "Define the taxi trip and fare predictions classes")]
+
+`TaxiTrip` is the input data class and has definitions for each of the data set columns. Use the [Column](xref:Microsoft.ML.Runtime.Api.ColumnAttribute) attribute to specify the indices of the source columns in the data set.
+
+The `TaxiTripFarePrediction` class is used to represent predicted results. It has a single float (`FareAmount`) field with a `Score` [ColumnName](xref:Microsoft.ML.Runtime.Api.ColumnNameAttribute) attribute applied. The **Score** column is the special column in ML.NET. The model outputs predicted values into that column.
```

Comment:
@aditidugar please check that the description of the **Score** column is correct. The current version of the tutorial doesn't explain why to use the `ColumnName` attribute with the `Score` name. However, it's quite important to have that attribute (I guess the field named `Score` also would work).

Reply:
@pkulikov that is correct. Either field named Score or attribute.